client/android/Crowd-ML/app/src/main/java/osu/crowd_ml/BackgroundDataSend.java

```diff
@@ -296,14 +278,14 @@ private void removeFirebaseListeners(){
 
     @Override public void onDestroy() {
 
+        Log.d("onDestroy", "Unregister Wifi receiver.");
+        // Step 1. End the wifi receiver.
+        unregisterReceiver(receiver);
```

Comment:
Stop the work thread before disabling wifi. Since the work thread will be interrupted by a wifi change, it's better practice to end the dependent component first.